### PR TITLE
Add initial README.md with examples

### PR DESCRIPTION
**Pull request by terrytangyuan**
Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,255 @@
...
+# Couler
+
+## What is Couler?
+
+Couler aims to provide a unified interface for constructing and managing workflows on
+different workflow engines, such as [Argo Workflows](https://github.com/argoproj/argo) and [Apache Airflow](https://airflow.apache.org/).
+
+## Why use Couler?
+
+Many workflow engines exist nowadays, e.g. [Apache Airflow](https://airflow.apache.org/), [Kubeflow Pipelines](https://github.com/kubeflow/pipelines), and [Argo Workflows](https://github.com/argoproj/argo).
```
Comment (ywskycn) [inline README.md:12]: Not sure whether we should treat kfp as an engine here, as it is a layer atop engines. One other example can be included here is Tekton.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,255 @@
...
+
+This example combines the use of a Python function result, along with conditionals,
+to take a dynamic path in the workflow. In this example, depending on the result
+of the first step defined in `flip_coin()`, the template will either run the
+`heads()` step or the `tails()` step.
+
+Steps can be defined via either `couler.run_script()`
+for Python functions or `couler.run_container()` for containers. In addition,
+the conditional logic to decide whether to flip the coin in this example
+is defined via the combined use of `couler.when()` and couler.equal()`.
```
Comment (ywskycn) [inline README.md:136]: This is a \` missing before `couler.equal()`

File: README.md [hunk 3]
```diff
@@ -1,3 +1,255 @@
...
+    )
+
+
+def tails():
+    return couler.run_container(
+        image="alpine:3.6", command=["sh", "-c", 'echo "it was tails"']
+    )
+
+
+couler.config_workflow(timeout=3600, time_to_clean=3600 * 1.5)
```
Comment (ywskycn) [inline README.md:166]: we can remove this line, as we haven't talked about this feature yet.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,255 @@
...
+            [lambda: job_b(message="C"), lambda: job_d(message="D")],  # C -> D
+        ]
+    )
+
+
+couler.config_workflow(timeout=3600, time_to_clean=3600 * 1.5)
+linear()
+submitter = ArgoSubmitter(namespace="argo")
+couler.run(submitter=submitter)
+```
```
Comment (ywskycn) [inline README.md:255]: We can add some words below to welcome discussions and contributions

File: README.md [hunk 5]
```diff
@@ -1,3 +1,255 @@
...
+            [lambda: job_a(message="A")],
+            [lambda: job_a(message="A"), lambda: job_b(message="B")],  # A -> B
+            [lambda: job_a(message="A"), lambda: job_c(message="C")],  # A -> C
+            [lambda: job_b(message="B"), lambda: job_d(message="D")],  # B -> D
+            [lambda: job_b(message="C"), lambda: job_d(message="D")],  # C -> D
+        ]
+    )
+
+
+couler.config_workflow(timeout=3600, time_to_clean=3600 * 1.5)
```
Comment (ywskycn) [inline README.md:251]: same as above

**Comment by terrytangyuan** [comment]
@ywskycn Thanks for the great suggestions! Updated.

**Review by ywskycn — APPROVED** [review]

**Pull request merged**